Post-release User Docs review - Merge sections Reporting an Issue and Problem Analysis (#220)
trento-project/docs: trento/adoc/trento-report-issue.adoc

include::product-attributes.adoc[]

[[trento-report-issue]]
== Reporting an issue
:revdate: 2026-06-19

> To report a {trento} issue to support, first open a support case, collect the required diagnostic data from your {trserver} and gather any relevant {tragent} information.

Reviewer: ```suggestion
To report an issue with {trento}, open a support case through the appropriate channel, collect the required diagnostic data from {trserver} and the relevant {tragent}s, and attach all that information to your support case.
```

[[opening-support-case]]
=== Opening a support case

{suse} customers with registered {sles4sap}{nbsp}15 (SP4 or higher) or {sles4sap} 16 distributions can report {trento} issues either directly in the {scc}
or through the corresponding vendor, depending on their licensing model.

> * Report the problems under {sles4sap}{nbsp}15 and component trento.

Author: > * Report the problems under {sles4sap}{nbsp}15 and component trento.

-> Is there an option to report under 16?

Reviewer: Yes, there is:

<img width="682" height="412" alt="Image" src="https://github.com/user-attachments/assets/e9e21225-efc3-48ab-a02e-e2f6190a8fcd" />

> * Provide the relevant deployment option for {trserver}: {k8s}, or systemd deployment.

Reviewer: ```suggestion
* Specify the relevant deployment option for {trserver}: {k8s}, or systemd deployment.
```

* Depending on your deployment type, use the corresponding procedure to collect and provide either <<collecting-server-diagnostic-data-k3s, K3s server diagnostic data>> or <<collecting-server-diagnostic-data-systemd, systemd server diagnostic data>>.

* If the issue involves a particular {tragent} or a component discovered by it, you must also provide <<collecting-agent-diagnostic-data, {tragent} diagnostic data>>.

[[collecting-server-diagnostic-data-k3s]]
=== Collecting {trserver} diagnostic data: K3s deployments

For K3s ({k8s}) deployments, provide the output of the {trento} support script, a scenario dump, pod descriptions and pod logs.

> Before proceeding, ensure your environment meets the following requirements:

Reviewer: ```suggestion
Before proceeding, ensure your host meets the following requirements:
```

> * A {sles4sap} 15 SP4 or higher is installed.

Author: > * A {sles4sap} 15 SP4 or higher is installed.

Note: This was originally SP3 or higher, but since we say at the beginning that only customers with SP4 and newer can open a support case, plus SP4 and higher is supported, I changed it to SP4 here.

* Packages `jq` and `yq` are installed.
* Helm is installed.
* `kubectl` is installed and connected to the {k8s} cluster where {trserver} is running.

[.procedure]

. Use the {trento} support plug-in to automate the collection of logs and relevant runtime information on the server side.

.. Install the plug-in package:
+
[source,subs="attributes"]
----
{prompt_root}zypper ref

> {prompt_root}zypper install supportutils-plugin-trento

Reviewer: ```suggestion
zypper ref
zypper install supportutils-plugin-trento
```

----

.. Run the `trento-support.sh` script:
+
[source,subs="attributes"]
----

> {prompt_root}trento-support --output file-tgz --collect all

Reviewer: ```suggestion
trento-support --output file-tgz --collect all
```

----
+
[NOTE]
====
The script accepts the following options:

* `-o`, `--output` Output type (`stdout`, `file`, `file-tgz`)
* `-c`, `--collect`  Collection options (`configuration`, `base`, `kubernetes`, `all`)
* `-r`, `--release-name` Release name to use for the chart installation. Default is `trento-server`
* `-n`, `--namespace` {k8s} namespace used when installing the chart. Default is `default`
* `--help` Shows help messages
====

.. Save the generated archive file to send to support.

. Generate a scenario dump of the database to help the {trento} team re-create and test the scenario.

.. Download the latest version of the dump script from the {trento} upstream project:
+
[source,subs="attributes"]
----

> {prompt_user}wget https://raw.githubusercontent.com/trento-project/web/main/hack/dump_scenario_from_k8.sh

Reviewer: ```suggestion
wget https://raw.githubusercontent.com/trento-project/web/main/hack/dump_scenario_from_k8.sh
```

----

.. Make the script executable:
+
[source,subs="attributes"]
----

> {prompt_user}chmod +x dump_scenario_from_k8.sh

Reviewer: ```suggestion
chmod +x dump_scenario_from_k8.sh
```

----

.. Run the script:
+
[source,subs="attributes"]
----

> {prompt_user}./dump_scenario_from_k8.sh --name SCENARIO_NAME --path PATH

Reviewer: ```suggestion
./dump_scenario_from_k8.sh --name SCENARIO_NAME --path PATH
```

----

.. Package all generated JSON files found under `PATH/scenarios/SCENARIO_NAME` to send to support.

. Collect pod descriptions and logs to provide additional context and details for support.

.. List the pods running in {k8s} cluster to view their statuses and names:
+
[source,subs="attributes"]
----

> {prompt_user}kubectl get pods

Reviewer: ```suggestion
kubectl get pods
```

NAME                                               READY   STATUS    RESTARTS   AGE
trento-server-postgresql-0                         1/1     Running   0          87m
trento-server-prometheus-server-7b5c9474bc-wmv8s   2/2     Running   0          87m
trento-server-rabbitmq-0                           1/1     Running   0          87m
trento-server-wanda-67ffbb79dc-7t6xw               1/1     Running   0          87m

> trento-server-web-7df8f65794-vdbhs                 1/1     Running   0          87m

Reviewer: ```suggestion
```

----

.. Retrieve the description of a pod:
+
[source,subs="attributes"]
----

> {prompt_user}kubectl describe pod POD_NAME

Reviewer: ```suggestion
kubectl describe pod POD_NAME
```

----

.. Retrieve the log of a pod:
+
[source,subs="attributes"]
----

> {prompt_user}kubectl logs POD_NAME

Reviewer: ```suggestion
kubectl logs POD_NAME
```

----

.. Monitor the log of a pod if required:
+
[source,subs="attributes"]
----

> {prompt_user}kubectl logs POD_NAME --follow

Reviewer: ```suggestion
```

----

.. Save the generated pod descriptions and log outputs to text files to send to support.

[[collecting-server-diagnostic-data-systemd]]
=== Collecting {trserver} diagnostic data: systemd deployments

For systemd environments, provide the output of the {trento} support plug-in, which automates the collection of logs and relevant runtime information on the server side.

[.procedure]

. Install the plug-in package:
+
[source,subs="attributes"]
----
{prompt_root}zypper ref

> {prompt_root}zypper install supportutils-plugin-trento

Reviewer: ```suggestion
zypper ref
zypper install supportutils-plugin-trento
```

----

. Run the `supportconfig` command to call the plugin:
+
[source,subs="attributes"]
----
{prompt_root}supportconfig
----
+

> For more information on how to run this command, see https://documentation.suse.com/smart/systems-management/html/supportconfig/index.html.

Reviewer: ```suggestion
. Run the `supportconfig` command to call the plugin. For more information on how to run this command, see https://documentation.suse.com/smart/systems-management/html/supportconfig/index.html.
```
